Cargo.toml

```diff
@@ -36,6 +36,10 @@ toml = "0.1"
 url = "0.2"
 winapi = "0.1"
 
+[dependencies.file-lock]
+git = "https://github.com/Byron/file-lock"
+branch = "cargo-adjustments"
```

Comment:
Absolutely - my preferred path of action is to get the cargo-related integration right, then port file-lock to windows, and see if cargo indeed works and builds on all supported platforms. Probably you can ask *bors* to do that, as travis can't test windows for instance.

Once it does work everywhere, a new crates.io release of the original crate should be made (I just have a fork).